schedulingApp/unit-tests/test_courses.py

```diff
@@ -1,38 +1,131 @@
 from django.test import TestCase
-
+from models import Course
 
 # Fields: Title
 class CreateCourse(TestCase):
-    def test_emptyTitle(self):
-        pass
+    def test_noArgs(self):
+        with self.assertRaises(TypeError, msg="Should raise TypeError for no arguments, requires a title and semester"):
+            a = Course()
 
-    def test_conflictingTitle(self):
-        pass
+    def test_oneArg(self):
+        with self.assertRaises(TypeError, msg="Should raise TypeError for only one argument, requires a title and semester"):
+            a = Course("CS361")
 
     def test_validInfo(self):
-        pass
+        a = Course("CS361", "Fall 2022")
+        self.assertEqual(a.title, "CS361", msg="Course did not successfully create with a valid title.")
+        self.assertEqual(a.semester, "Fall 2022", msg="Course did not successfully create with a valid semester")
+
+    def test_threeArgs(self):
+        with self.assertRaises(TypeError, msg="Should raise TypeError for three arguments, should only take two."):
+            a = Course("CS361", "Fall 2022", "Spring 2023")
+
+    def test_emptyTitle(self):
+        with self.assertRaises(ValueError, msg="Should raise ValueError for empty string title, should require a non-empty title"):
+            a = Course("", "Fall 2022")
+
+    def test_emptySemester(self):
+        with self.assertRaises(ValueError, msg="Should raise ValueError for empty string semester, should require a "
+                                               "non-empty semester"):
+            a = Course("CS361", "")
+
+    def test_conflictingCourse(self):
+        a = Course("CS361", "Fall 2022")
+        with self.assertRaises(ValueError, msg="Course with duplicate name and semester should not be allowed."):
+            b = Course("CS361", "Fall 2022")
 
 
 class EditCourse(TestCase):
+    theCourse = None
+
+    def setUp(self):
+        theCourse = Course("CS361", "Fall 2022")
+        theCourse.title = "CS361"
+        theCourse.semester = "Fall 2022"
+
+    #testing title changes
     def test_blankTitle(self):
-        pass
+        self.setUp()
```

Comment:
You can remove the self.setUp() calls as TestCase calls that automatically before each method is called